### DISCUSSION TITLE
Role-based authorization for OpenID provider

### DISCUSSION THREAD

**Discussion by siriusfreak** [danny-avila/LibreChat#1491] [Feature Requests & Suggestions]
Hello.

Before reading the Contribution guide, I implemented a small change that allows to perform role-based authorization for OpenID provider.

Draft of pull request with changes: https://github.com/danny-avila/LibreChat/pull/1488 

Do you think this change could be helpful?

**Comment by danny-avila** [comment]
Looks good to me, only qualm I have is adding the extra jwt package. Can you accomplish the same through the already installed package?

`const jwt = require('jsonwebtoken');`

**Reply by danny-avila** [reply]
And making sure OPENID functions as expected without ever setting the role variable
